tensorflow/core/kernels/cwise_ops.h

```diff
@@ -236,6 +236,110 @@ struct functor_traits<scalar_compose_op<Scalar, UnaryFunctor, BinaryFunctor>> {
   };
 };
 
+// tag for not complex number pass to tanh
+struct tanh_not_complex_tag {};
+// tag for complex number pass to tanh
+struct tanh_complex_tag {};
+// struct for tag dispatching
+template <typename T>
+struct tanh_dispatching_tag {
+  typedef tanh_not_complex_tag type;
+};
+// struct specify for complex<double>
+template <>
+struct tanh_dispatching_tag<std::complex<double>> {
+  typedef tanh_complex_tag type;
+};
+// struct specify for complex<float>
+template <>
+struct tanh_dispatching_tag<std::complex<float>> {
+  typedef tanh_complex_tag type;
+};
+// compute tanh for complex number
+template <typename T>
+const T tanh_dispatching(tanh_complex_tag, const T& _x) {
+  return numext::tanh(_x);
+}
+// compute tanh for not complex number
+template <typename T>
+const T tanh_dispatching(tanh_not_complex_tag, const T& _x) {
```

Comment:
It would be a simpler and we wouldn't loose the vectorized implementation if you instead fixed the eigen code to return consistent results with a without vectorization. How about updating https://bitbucket.org/eigen/eigen/src/c44a78394f3f9d9bfc75caad97feb14c4183c942/Eigen/src/Core/functors/UnaryFunctors.h?at=default&fileviewer=file-view-default#UnaryFunctors.h-501 to use the code you implemented in tanh_dispatching instead of relying on std::tanh?